homeassistant/components/bkk/sensor.py

```diff
@@ -0,0 +1,314 @@
+            return
+        entities = []
+        stop_type = data_updater.data["references"]["stops"][data_updater.stop_id][
+            "type"
+        ]
+        for i in range(MAX_DEPARTURES):
+            route = BkkRoute(data_updater.stop_id, i, stop_type)
+            entities.append(route)
+        data_updater.routes = entities
+        async_add_entities(entities, True)
+        async_add_entities([data_updater], True)
+
+        async def async_remove_unavailable_entities(event):
+            reg = await entity_registry.async_get_registry(hass)
+            entities = reg.entities.copy()
+            for ent in entities:
+                registry_entry = reg.async_get(ent)
+                if registry_entry.platform == DOMAIN:
+                    entity_id = registry_entry.entity_id
+                    state = hass.states.get(entity_id).state
+                    if state == STATE_UNAVAILABLE:
+                        reg.async_remove(entity_id)
+
+        hass.bus.async_listen_once(
+            EVENT_HOMEASSISTANT_START, async_remove_unavailable_entities
+        )
+        return True
+
+
+class BkkDataUpdater(Entity):
+    """Data updater class to query api and update entity data."""
+
+    def __init__(self, config):
+        """Initialize the sensor."""
+        self._name = "BKK Data Updater"
+        self.api = futar.BkkFutar()
+        self.routes = None
+        self.stop_id = (
+            f"BKK_{config[CONF_STOP][1:]}"
+            if config[CONF_STOP][0] == "#"
+            else f"BKK_{config[CONF_STOP]}"
+        )
+        self._data = None
+        self._state = None
+        self._icon = ICONS["API"]
+        self.is_grouped = config[CONF_GROUPED]
+        self._only_departures = config[CONF_ONLY_DEPARTURES]
+        self._minutes_after = config[CONF_MINUTES_AFTER]
+
+    def get_data(self):
+        """Fetch data from BKK api."""
+        try:
+            response = self.api.arrivals_and_departures_for_stop(
+                stopid=self.stop_id,
+                only_departures=self._only_departures,
+                minutes_after=self._minutes_after,
+            )
+        except ConnectionError:
+            _LOGGER.error("Unable to fetch data from api.")
+        else:
+            status = response["status"]
+            if status == "OK":
+                self._data = response["data"]
+            self._state = status
+            return self._data
+
+    async def async_added_to_hass(self) -> None:
+        """Create group after entity added."""
+        entity_ids = [ent.entity_id for ent in self.routes]
+        stop_name = self.data["references"]["stops"][self.stop_id]["name"]
+        if self.is_grouped:
+            await async_create_group(self.hass, stop_name, entity_ids)
+
+    @property
+    def state(self):
+        """Get BKKDataUpdater state."""
+        return self._state
+
+    @property
+    def hidden(self) -> bool:
+        """Set BKKDataUpdater hidden."""
+        return True
```

Comment:
Hidden will be deprecated in `0.107`, see [advanced properties](https://developers.home-assistant.io/docs/en/entity_index.html#advanced-properties).

Reply:
removed